cmd/rqlite/query.go

```diff
@@ -80,83 +80,56 @@ type queryResponse struct {
 	Time    float64 `json:"time"`
 }
 
-func queryWithClient(ctx *cli.Context, client *http.Client, argv *argT, timer bool, consistency, query string) error {
+func queryWithClient(ctx *cli.Context, client *cl.Client, timer bool, consistency, query string) error {
 	queryStr := url.Values{}
 	queryStr.Set("level", consistency)
 	queryStr.Set("q", query)
 	if timer {
 		queryStr.Set("timings", "")
 	}
 	u := url.URL{
-		Scheme:   argv.Protocol,
-		Host:     fmt.Sprintf("%s:%d", argv.Host, argv.Port),
-		Path:     fmt.Sprintf("%sdb/query", argv.Prefix),
+		Path:     fmt.Sprintf("%sdb/query", client.Prefix),
 		RawQuery: queryStr.Encode(),
 	}
-	urlStr := u.String()
 
-	nRedirect := 0
-	for {
-		req, err := http.NewRequest("GET", urlStr, nil)
-		if err != nil {
-			return err
-		}
-		if argv.Credentials != "" {
-			creds := strings.Split(argv.Credentials, ":")
-			if len(creds) != 2 {
-				return fmt.Errorf("invalid Basic Auth credentials format")
-			}
-			req.SetBasicAuth(creds[0], creds[1])
-		}
-
-		resp, err := client.Do(req)
-		if err != nil {
-			return err
-		}
-		response, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return err
-		}
-		resp.Body.Close()
-
-		if resp.StatusCode == http.StatusUnauthorized {
-			return fmt.Errorf("unauthorized")
-		}
+	resp, err := client.Query(u)
+	if err != nil {
+		return err
+	}
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	resp.Body.Close()
 
-		if resp.StatusCode == http.StatusMovedPermanently {
-			nRedirect++
-			if nRedirect > maxRedirect {
-				return fmt.Errorf("maximum leader redirect limit exceeded")
-			}
-			urlStr = resp.Header["Location"][0]
-			continue
-		}
+	if resp.StatusCode == http.StatusUnauthorized {
```

Comment:
Same as earlier, not sure this check adds anything versus the check immediately afterwards.